### PR TITLE
[dmd.root.string]: Address reviewer feedback

### PR DESCRIPTION
**Pull request by PetarKirov**
Address reviewer feedback from #11015 

FWIW, I still disagree with these changes, but I don't want spend time debating.

### CODE REVIEW

File: src/dmd/root/string.d [hunk 1]
```diff
@@ -269,7 +269,8 @@ do
 {
     foreach (const c; needle)
     {
-        if (!*p || c != *p)
+        assert(*p);
```
Comment (Geod24) [inline src/dmd/root/string.d:272]:
```suggestion
        assert(c);
```

File: src/dmd/root/string.d [hunk 2]
```diff
@@ -227,7 +227,7 @@ unittest
  *    - https://gist.github.com/PetarKirov/338e4ab9292b6b2b311a3070572a07fb (backup URL)
 */
 T[N + 1] toStaticArray(T, size_t N)(scope const(T)[N] literal)
-if (is(T == char) || is(T == wchar) || is(T == dchar))
+if (is(T == char))
```
Comment (WalterBright) [inline src/dmd/root/string.d:230]:
```
char[N + 1] toStaticArray(size_t N)(scope const(char)[N] literal)
```

**Comment by WalterBright** [comment]
Thank you, @PetarKirov

File: src/dmd/root/string.d (hunk 2 above)
Comment (PetarKirov) [reply src/dmd/root/string.d]: Done

**Review by Geod24 — APPROVED** [review]

**Pull request merged**